### ISSUE TITLE
Incorrect validation rule: is_string() on int[] values

### ISSUE THREAD

**Issue by GroxExMachine** [WsdlToPhp/PackageGenerator#60]
WSDL: https://api.direct.yandex.ru/live/v4/wsdl/
```php
    /**
     * Set GeoID value
     * @throws \InvalidArgumentException
     * @param int[] $geoID
     * @return \ApiV4Live\StructType\NewForecastInfo
     */
    public function setGeoID(array $geoID = array())
    {
        foreach ($geoID as $newForecastInfoGeoIDItem) {
            // validation for constraint: itemType
            if (!is_string($newForecastInfoGeoIDItem)) {
                throw new \InvalidArgumentException(sprintf('The GeoID property can only contain items of int, "%s" given', is_object($newForecastInfoGeoIDItem) ? get_class($newForecastInfoGeoIDItem) : gettype($newForecastInfoGeoIDItem)), __LINE__);
            }
        }
        $this->GeoID = $geoID;
        return $this;
    }
```

**Comment by GroxExMachine** [comment]
```php
    /**
     * Add item to GeoID value
     * @throws \InvalidArgumentException
     * @param int $item
     * @return \AppBundle\Provider\YandexDirect\ApiV4Live\StructType\NewForecastInfo
     */
    public function addToGeoID($item)
    {
        // validation for constraint: itemType

        if (!is_string($item)) {
            throw new \InvalidArgumentException(sprintf('The GeoID property can only contain items of int, "%s" given', is_object($item) ? get_class($item) : gettype($item)), __LINE__);
        }
        $this->GeoID[] = $item;
        return $this;
    }
```

**Comment by mikaelcom** [comment]
Does it raise a notice or an error? Or is it mainly the validation rule that should be better by using `is_numeric` instead of `is_string` as in [IntRule](https://github.com/WsdlToPhp/PackageGenerator/blob/develop/src/File/Validation/IntRule.php)?

**Comment by GroxExMachine** [comment]
It raises an Exception
  [InvalidArgumentException]
  The GeoID property can only contain items of int, "integer" given

**Comment by mikaelcom** [comment]
Actually it is weird. Which version are you using?

I'm going to publish an improvement that does not fundamentaly change the validation rule. The unit tests already contained your case and it was working well. Indeed, int/long values should be verified with `is_numeric` instead of `is_int`as in this commit https://github.com/WsdlToPhp/PackageGenerator/commit/0c0e5d34183ec371206af659841bba01fb59fa4d. In your case, with my version, it generated `is_int` validations instead of `is_string`.

**Comment by GroxExMachine** [comment]
> Which version are you using?
I used the last one from 1.x branch. Looked like 1.5.1.

With the most fresh from dev branch I have no errors for now.

Thank you for fast update!

**Comment by mikaelcom** [comment]
fixed in [1.7.0](https://github.com/WsdlToPhp/PackageGenerator/releases/tag/1.7.0) and [2.2.0](https://github.com/WsdlToPhp/PackageGenerator/releases/tag/2.2.0)
